### PR TITLE
Implement option to start at a specific exercise

### PR DESCRIPTION
**Pull request by louis77**
A humble first try to tackle issue #156 :-)

### CODE REVIEW

**Comment by danwritecode** [comment]
Thinking about this one more, I think we probably should support this feature in both normal and debug modes. Debug mode doesn't allow you to fully test (at least right now) and if someone is developing a new exercise they need to be able to actually run it. 

Do you mind updating? Sorry about that, didn't think of it when I was creating the issue.

**Comment by louis77** [comment]
Agree. I've removed the 2nd commit, so now it's available in non-debug mode, too. That's probably also more useful if more CLI flags are added in the future.

File: src/main.c [hunk 1]
```diff
@@ -21,10 +21,35 @@ FailureMode find_failure_mode(char *file_path, char *file_name_no_ext);
...
 
+    int start_at = 1;
+    int c;
+
+    opterr = 0;
+
+    while ((c = getopt(argc, argv, "s::")) != -1) {
+        switch (c) {
+            case 's':
+                start_at = atoi(optarg);
```
Comment (danwritecode) [inline src/main.c:37]: Could you add a check here to make sure optarg isn't null? if it is null, then we should display some message that the -s requires a number to start at

**Review by danwritecode — CHANGES_REQUESTED** [review]
Looks good, just want to add a check for optarg.

File: src/main.c [hunk 2]
```diff
@@ -21,10 +21,35 @@ FailureMode find_failure_mode(char *file_path, char *file_name_no_ext);
...
-int main() {
+int main(int argc, char *argv[]) {
     signal(SIGINT, sig_handler);
 
+    int start_at = 1;
+    int c;
+
+    opterr = 0;
+
+    while ((c = getopt(argc, argv, "s::")) != -1) {
```
Comment (danwritecode) [inline src/main.c:34]: should this be "s:" this way the argument can be passed with a space?

**Review by danwritecode — CHANGES_REQUESTED** [review]

**Review by danwritecode — APPROVED** [review]

**Pull request merged**